src/coreclr/jit/lowerxarch.cpp

```diff
@@ -7819,13 +8056,31 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
 
                     if (IsContainableHWIntrinsicOp(node, op2, &supportsOp2RegOptional))
                     {
-                        MakeSrcContained(node, op2);
+                        if (op2->OperIs(GT_CNS_VEC) && op2->TypeIs(TYP_SIMD16, TYP_SIMD32, TYP_SIMD64) &&
```

Comment:
Do we need the `op2->TypeIs(TYP_SIMD16, TYP_SIMD32, TYP_SIMD64)` check? Any broadcast vector constant should be fine to handle here, right?

Reply:
right this point, yes, it was intended to filter out `SIMD8/12`, but it should be already handled above.